pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1907,11 +1907,16 @@ public void seek(long timestamp) throws PulsarClientException {
 
         log.info("[{}][{}] Seek subscription to publish time {}", topic, subscription, timestamp);
 
-        cnx.sendRequestWithId(seek, requestId).thenRun(() -> {
+        cnx.sendSeekRequestWithId(seek, requestId).thenAccept((seekResponse) -> {
             log.info("[{}][{}] Successfully reset subscription to publish time {}", topic, subscription, timestamp);
             acknowledgmentsGroupingTracker.flushAndClean();
 
-            seekMessageId = new BatchMessageIdImpl((MessageIdImpl) MessageId.earliest);
+            MessageIdData messageIdData = seekResponse.getMessageIdData();
+            seekMessageId = new BatchMessageIdImpl(messageIdData.getLedgerId(), messageIdData.getEntryId(),
+                    messageIdData.getPartition(), -1);
```

Comment:
Why set the batch index to -1? And in `seekAsync(MessageId messageId)`, `seekMessageId` was set the `messageId` but not what the seek response contains.

Reply:
Batch index set to -1 since currently broker find message by timestamp at entry level.
I think we can raise up a new issue if we want to  support it at messag level.